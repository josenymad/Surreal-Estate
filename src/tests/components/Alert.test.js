import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Alert from "../../components/Alert";

describe("Alert", () => {
  it("displays an error message", () => {
    render(
      <MemoryRouter>
        <Alert message="Error!" />
      </MemoryRouter>
    );

    const alertMessageNode = screen.getByText(/Error/);

    expect(alertMessageNode.textContent).toBe("Error!");
  });

  it("displays a success message", () => {
    render(
      <MemoryRouter>
        <Alert message="Success!" success />
      </MemoryRouter>
    );

    const alertMessageNode = screen.getByText(/Success/);

    expect(alertMessageNode.textContent).toBe("Success!");
  });

  it("does not render if there is no message", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Alert message="" />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
