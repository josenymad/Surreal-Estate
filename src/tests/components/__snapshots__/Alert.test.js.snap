// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert does not render if there is no message 1`] = `<DocumentFragment />`;
